### ISSUE TITLE
dart format --set-exit-if-changed not really setting the status code

### ISSUE THREAD

**Issue by llucax** [dart-lang/sdk#44582]
I'm not 100% sure the `dart` cli tool, but I've seen other issues with it reported here, so I guess it it.

When I use `dart format --output none --set-exit-if-changed` the exit status is not really changed (it used 0):

```console
$ dart --version
Dart SDK version: 2.12.0-133.2.beta (beta) (Tue Dec 15 09:55:09 2020 +0100) on "linux_x64"
$ dart create bug8
Creating /tmp/bug8 using template console-simple...

  .gitignore
  CHANGELOG.md
  README.md
  analysis_options.yaml
  bin/bug8.dart
  pubspec.yaml

Running pub get...                     0.9s
  Resolving dependencies...
  Changed 2 dependencies!

Created project bug8! In order to get started, type:

  cd bug8

$ cd bug8/
$ sed -i '2s/ //g' bin/bug8.dart 
$ cat bin/bug8.dart 
void main(List<String> arguments) {
print('Helloworld!');
}
$ dartfmt --set-exit-if-changed --dry-run .
bin/bug8.dart
$ echo $? # This works as it should
1
$ dart format --set-exit-if-changed --output none .
Changed bin/bug8.dart
Formatted 1 file (1 changed) in 0.12 seconds.
$ echo $? # This doesn't!
0
$ 
```

**Comment by munificent** [comment]
dart_style should be setting the exit code correctly. I have integration tests in the package to verify. It sets the exit code [here](https://github.com/dart-lang/dart_style/blob/master/lib/src/cli/formatter_options.dart#L79) using the `exitCode` setter in dart:io. Is it possible that the CLI integration is overriding that?

**Comment by munificent** [comment]
From talking to @devoncarew, it seems likely that this has to do with the integration into Dart CLI, so assigning it back.

**Comment by bkonyi** [comment]
Yeah, we don't use `exitCode` since we need to pass the value back to the VM through the CLI isolate. Should be an easy enough fix.

**Comment by bkonyi** [comment]
This has been fixed in `dart_style` and just needs to be rolled into the SDK. Reassigning to @munificent to complete the roll, and I'll add testing for the CLI once this happens.

**Comment by munificent** [comment]
Patch to roll it into the SDK is out for review: https://dart-review.googlesource.com/c/sdk/+/187740

**Comment by kuhnroyal** [comment]
This was merged in https://github.com/dart-lang/sdk/commit/3f5384bd084cdd15a7cc8671f6110e5a25b10080 - will there be a 2.12.x patch release with this?

**Comment by munificent** [comment]
This is in the SDK now, so bouncing to @bkonyi.

**Comment by vasilich6107** [comment]
Checking in flutter 2.5.1 
`flutter format --set-exit-if-changed lib` - works fine and throws error code 1
`dart format --set-exit-if-changed lib` - ignores `--set-exit-if-changed` and always throws error code 0

**Comment by munificent** [comment]
I can't repro what you're seeing with the latest Dart SDK:

```
$ dart format --set-exit-if-changed temp.dart
Formatted temp.dart
Formatted 1 file (1 changed) in 0.15 seconds.
$ echo :$?
:1
```

I'm not 100% certain that this fix is in Flutter 2.5.1, but I would be surprised if it wasn't. Is it possible that you have an older Dart SDK on your PATH somehow?

**Comment by vasilich6107** [comment]
Maybe)
Until it works with `flutter format` it’s ok for me)
Thanks.
